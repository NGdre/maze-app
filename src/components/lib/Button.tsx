import React from "react";

const btnClassName = `py-2.5 px-5 me-2 text-sm font-medium text-gray-900 bg-white
           rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700
            focus:z-10 focus:ring-4 focus:outline-none focus:ring-blue-700
             focus:text-blue-700 inline-flex items-center`;

type btnType = "button" | "submit" | "reset" | undefined;

const defaultBtnType = "button";

// this component exist so you don't have to write className's content in every button
export default function Button({
  type,
  onClick,
  disabled = false,
  children,
}: {
  type?: btnType;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
  children: React.ReactNode;
}) {
  return (
    <button
      type={type || defaultBtnType}
      onClick={onClick}
      disabled={disabled}
      className={btnClassName}
    >
      {children}
    </button>
  );
}
