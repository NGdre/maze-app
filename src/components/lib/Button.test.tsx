import { render } from "@testing-library/react";
import Button from "./Button";

describe(Button.name, () => {
  it("should take a snapshot", () => {
    const clickHandler = jest.fn();

    const { asFragment } = render(
      <Button onClick={clickHandler}>click me</Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
