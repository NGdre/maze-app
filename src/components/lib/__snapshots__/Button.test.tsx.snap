// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="py-2.5 px-5 me-2 text-sm font-medium text-gray-900 bg-white
           rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700
            focus:z-10 focus:ring-4 focus:outline-none focus:ring-blue-700
             focus:text-blue-700 inline-flex items-center"
    type="button"
  >
    click me
  </button>
</DocumentFragment>
`;
